### DISCUSSION TITLE
Rendering small tif file

### DISCUSSION THREAD

**Discussion by julianblue** [developmentseed/titiler#812] [Q&A]
Hi, 

I am testing titiler endpoints to visualise clipped satellite images. These files can have dimensions as low as 150x150 pixels. When using the tile endpoints the images are not fully rendered as seen below. Is the issue linked to the cog configuration or something else ?
(Black areas render as soon as I zoom in a lot closer.)

<img width="1147" alt="Screenshot 2024-04-23 at 09 06 42" src="https://github.com/developmentseed/titiler/assets/21318315/6344b299-f5e4-4303-a992-99dde8bb8167">

rio cogeo info 
bash`
Profile
    Width:            71
    Height:           173
    Bands:            8
    Tiled:            True
    Dtype:            uint16
    NoData:           0.0
    Alpha Band:       False
    Internal Mask:    False
    Interleave:       PIXEL
    ColorMap:         False
    ColorInterp:      ('undefined', 'blue', 'undefined', 'green', 'undefined', 'red', 'undefined', 'undefined')
    Scales:           (1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)
    Offsets:          (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)

Geo
    Crs:              EPSG:32632
    Origin:           (518409.0, 5503611.0)
    Resolution:       (3.0, -3.0)
    BoundingBox:      (518409.0, 5503092.0, 518622.0, 5503611.0)
    MinZoom:          15
    MaxZoom:          15

Image Metadata
    AREA_OR_POINT: Area
    TIFFTAG_DATETIME: 2024:04:21 09:36:55
    TIFFTAG_IMAGEDESCRIPTION: {"atmospheric_correction": {"aerosol_model": "continental", "aot_coverage": 1.0, "aot_mean_quality": 4, "aot_method": "map", "aot_source": "mod09cma_nrt", "aot_status": "Data Found", "aot_std": 0.0, "aot_used": 0.06, "atmospheric_correction_algorithm": "6Sv2.1", "atmospheric_model": "water_vapor_and_ozone", "luts_version": 3, "ozone_coverage": 1.0, "ozone_mean_quality": 260, "ozone_method": "fixed", "ozone_source": "mod09cmg_nrt", "ozone_status": "Data Found", "ozone_std": 0.0005, "ozone_used": 0.44, "satellite_azimuth_angle": 0, "satellite_zenith_angle": 0, "solar_azimuth_angle": 140.929, "solar_zenith_angle": 43.335, "sr_version": "2.1", "water_vapor_coverage": 1.0, "water_vapor_mean_quality": 4, "water_vapor_method": "fixed", "water_vapor_source": "mod09cma_nrt", "water_vapor_status": "Data Found", "water_vapor_std": 0.073, "water_vapor_used": 0.51}}

Image Structure
    COMPRESSION: LZW
    INTERLEAVE: PIXEL
    LAYOUT: COG
    PREDICTOR: 2

Band 1
    Description: coastal_blue
    ColorInterp: undefined

Band 2
    Description: blue
    ColorInterp: blue

Band 3
    Description: green_i
    ColorInterp: undefined

Band 4
    Description: green
    ColorInterp: green

Band 5
    Description: yellow
    ColorInterp: undefined

Band 6
    Description: red
    ColorInterp: red

Band 7
    Description: rededge
    ColorInterp: undefined

Band 8
    Description: nir
    ColorInterp: undefined

IFD
    Id      Size           BlockSize     Decimation
    0       71x173         512x512       0
`

Thanks in advance.

**Comment by vincentsarago** [comment]
@julianblue interesting, the nodata is set to `0` but there might be something going on with rio-tiler. are you able to share your file?

**Comment by julianblue** [comment]
Hi @vincentsarago , 

sure here is an example file. Is the visualisation of small files recommended through TileJSON or another approach more suitable ?
[test.tif.zip](https://github.com/developmentseed/titiler/files/15099520/test.tif.zip)
